catkin_ws/src/state_estimation/src/state_aggregator.py

```diff
@@ -43,57 +67,95 @@ def __init__(self):
 
         # subscribers
         rospy.Subscriber("/sbg/ekf_quat", SbgEkfQuat, self.imu_cb)
+        rospy.Subscriber("/dead_reckon_report",DeadReckonReport, self.dvl_cb)
         rospy.Subscriber("/depth", Float64, self.depth_sensor_cb)
-        rospy.Subscriber("imu_reset", Empty, self.imu_reset_cb)
-        rospy.Subscriber("dead_reckon_report",DeadReckonReport, self.dr_cb)
-        rospy.Subscriber("[DVL_reset]", Empty, self.dvl_offset)
-
-        '''
-        TODO - use tf2 instead of publishing pose?
-        # transform broadcasters
-        self.br_world = TransformBroadcaster()
-        self.br_auv_base = StaticTransformBroadcaster()
-        
-        # transform messages
-        self.tf_world = TransformStamped()
-        self.tf_auv_base = TransformStamped()
-        '''
+        rospy.Subscriber("/reset_state", Empty, self.reset_state_cb)
+
+
+    def dvl_cb(self,data):
+        # quaternion of DVL in NED frame
+        q_dvl = transformations.quaternion_from_euler(data.roll, data.pitch, data.yaw)
+        # transformations returns quaternion as nparray [w, x, y, z]
+        q_dvl = np.quaternion(q_dvl[3], q_dvl[1], q_dvl[2], q_dvl[3]) 
```

Comment:
Shouldn't this be q_dvl = np.quaternion(q_dvl[3], q_dvl[0], q_dvl[1], q_dvl[2])?

Reply:
my b